x/mongo/driver/errors.go

```diff
@@ -317,12 +322,14 @@ func (e Error) NotMaster() bool {
 			return true
 		}
 	}
-	return strings.Contains(e.Message, "not master")
+	hasNoCode := e.Code == 0
+	return hasNoCode && strings.Contains(e.Message, "not master")
 }
 
 // NamespaceNotFound returns true if this errors is a NamespaceNotFound error.
 func (e Error) NamespaceNotFound() bool {
-	return e.Code == 26 || e.Message == "ns not found"
+	hasNoCode := e.Code == 0
```

Comment:
I vaguely remember discussing whether or not we should change this error helper. What was the result of your investigation? Is this in scope for this ticket?

Reply:
See comment above; changes to this helper removed.